B/Birch/build_tarballs.jl

```diff
@@ -0,0 +1,49 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Birch"
+version = v"1.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/lawmurray/Birch.git", "a533990b59b04ef4cd20f8d4fe91c4a220df1cfa"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# install the driver
+cd ${WORKSPACE}/srcdir/Birch/driver/
+autoreconf -vi
+CPPFLAGS="-I${prefix}/include" ./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+
+# install license
+install_license LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = expand_cxxstring_abis(supported_platforms())
+
+# GLOB_NOMAGIC is not supported by musl
+filter!(p -> libc(p) != "musl", platforms)
+
+# Native support for Windows is not yet provided
+filter!(!Sys.iswindows, platforms)
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("birch", :birch),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[
+    Dependency("boost_jll"),
```

Comment:
Do you need Boost libraries at _runtime_?  If not, this should be
```suggestion
    BuildDependency("boost_jll"),
```

Reply:
I am pretty sure it is needed (since Birch is a compiler we compile at runtime which requires additional libraries such as (maybe) boost) but I asked Lawrence in https://github.com/JuliaPackaging/Yggdrasil/pull/2156#discussion_r528204842 to have a look at it to ensure that I did not introduce any superfluous or missing dependencies when splitting the package.

Reply:
Short answer is yes, these are needed at runtime. That's because Birch is a compiler, and runtime for a compiler looks a lot like build time for anything else, for obvious reasons. For example, we might ask: "does `gcc` need `stdio.h` at runtime?". I mean yes, it does, or it can't compile much of anything useful. Same goes for Birch wanting various headers at runtime, not just build time.

Your best guide here to dependencies @devmotion is probably the [Homebrew packages](https://github.com/lawmurray/homebrew-birch). 

[Open Build Service](https://build.opensuse.org/project/show/home:lawmurray:birch) as well, but if you dig in there focus on the `PKGBUILD` files and not anything else. They're the packaging for Arch Linux, which is quite a bit simpler, as, like Homebrew, it doesn't separate packages into library and header (*-devel) packages like most of the other Linux distributions. (Unless for Julia packages are separated like this too? I'm guessing not.)

The dependencies listed in both of those cases for build time and run time are correct. There's nothing superfluous being added unnecessarily. I would just translate them as is to Julia's packaging, and also separate into separate packages as there (and as you're now doing, e.g. `birch`, `libbirch`, `birch-standard`, etc, following whatever the naming conventions are for Julia).

Reply:
Regarding the names of the packages: the [main restriction or convention is that]((https://juliapackaging.github.io/BinaryBuilder.jl/dev/building/#Name)

> It should be the name of the upstream package, not for example that of a specific library or executable provided by it, even though they may coincide. The case of the name should match that of the upstream package.

Therefore currently I chose the name `Birch` for the compiler, `LibBirch` for the LibBirch library, and `Birch_Standard` for the standard library, based on how these parts are called in their respective README files. I tried to match the case of the names as well, the underscore is a bit arbitrary though.